assets/queries/terraform/azure/role_definition_allows_custom_role_creation/query.rego

```diff
@@ -0,0 +1,28 @@
+package Cx
+
+CxPolicy[result] {
+	resource := input.document[i].resource.azurerm_role_definition[name]
+
+	actions := resource.permissions.actions
+
+	allows_custom_roles_creation(actions)
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("azurerm_role_definition[%s].permissions.actions", [name]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("azurerm_role_definition[%s].permissions.actions does not allow custom role creation", [name]),
+		"keyActualValue": sprintf("azurerm_role_definition[%s].permissions.actions allows custom role creation", [name]),
+	}
+}
+
+customRole := "Microsoft.Authorization/roleDefinitions/write"
+
+allows_custom_roles_creation(actions) {
+	count(actions) == 1
+	options := {"*", customRole}
+	actions[0] == options[x]
+} else {
+	count(actions) > 1
+	actions[x] == customRole
+}
```

Comment:
```suggestion
allows_custom_roles_creation(actions) {
	options := {"*", customRole}
	actions[x] == customRole
}
```
Can this be simplified like this?

Reply:
When I implemented it, I thought of restricting because `*` will never be set when `count(actions) > 1`. However, maybe restricting is of low importance

Reply:
I too think it is of low importance, but maybe we can let it as it is, as it will not have a big impact overall